src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -251,7 +252,10 @@ public ResponseEntity<ProgrammingExercise> createProgrammingExercise(@RequestBod
         programmingExerciseService.validateNewProgrammingExerciseSettings(programmingExercise, course);
 
         // Check that only allowed athena modules are used
-        athenaApi.ifPresentOrElse(api -> api.checkHasAccessToAthenaModule(programmingExercise, course, ENTITY_NAME), () -> programmingExercise.setFeedbackSuggestionModule(null));
+        athenaApi.ifPresentOrElse(api -> api.checkHasAccessToAthenaModule(programmingExercise, course, ModuleType.FEEDBACK_SUGGESTIONS, ENTITY_NAME),
+                () -> programmingExercise.setFeedbackSuggestionModule(null));
+        athenaApi.ifPresentOrElse(api -> api.checkHasAccessToAthenaModule(programmingExercise, course, ModuleType.PRELIMINARY_FEEDBACK, ENTITY_NAME),
+                () -> programmingExercise.setPreliminaryFeedbackModule(null));
```

Comment:
Why is this snippet not wrapped by try/catch? I saw you wrapped the same snippet in `src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseExportImportResource.java`